pinot-controller/src/test/java/com/linkedin/pinot/controller/helix/core/realtime/SegmentCompletionTest.java

```diff
@@ -320,6 +320,28 @@ public void testHappyPath(long startTime) throws  Exception {
   }
 
   @Test
+  public void testWinnerOnTimeLimit() throws Exception {
+    SegmentCompletionProtocol.Response response;
+    Request.Params params;
+    segmentCompletionMgr._secconds = 10L;
+    params = new Request.Params().withInstanceId(s1).withOffset(s1Offset).withSegmentName(segmentNameStr)
+        .withReason(SegmentCompletionProtocol.REASON_TIME_LIMIT);
+    response = segmentCompletionMgr.segmentConsumed(params);
+    Assert.assertEquals(response.getStatus(), SegmentCompletionProtocol.ControllerResponseStatus.HOLD);
+  }
+
+  @Test
+  public void testWinnerOnRowLimit() throws Exception {
+    SegmentCompletionProtocol.Response response;
+    Request.Params params;
+    segmentCompletionMgr._secconds = 10L;
+    params = new Request.Params().withInstanceId(s1).withOffset(s1Offset).withSegmentName(segmentNameStr)
```

Comment:
Perhaps you can have somebody elese come in with time limit and lower offset, either as a separate test or part of this test.